File: app/src/main/java/MainActivity.java
package com.example.project1mp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private Button btnLinear, btnFrame, btnRelative, btnConstraint;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnLinear = findViewById(R.id.btnLinearLayout);
        btnFrame = findViewById(R.id.btnFrameLayout);
        btnRelative = findViewById(R.id.btnRelativeLayout);
        btnConstraint = findViewById(R.id.btnConstraintLayout);

        btnLinear.setOnClickListener(v -> startActivity(new Intent(this, LinearActivity.class)));
        btnFrame.setOnClickListener(v -> startActivity(new Intent(this, FrameActivity.class)));
        btnRelative.setOnClickListener(v -> startActivity(new Intent(this, RelativeActivity.class)));
        btnConstraint.setOnClickListener(v -> startActivity(new Intent(this, ConstraintActivity.class)));
    }
}

File: app/src/main/java/com/example/project1mp/MainActivity.kt
package com.example.project1mp // Sesuai namespace proyek "Project 1 MP"

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var tvTitle: TextView
    private lateinit var btnStart: Button
    private lateinit var btnToggle: Button
    private lateinit var btnOpenPractice: Button
    private var isHello = true

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main) // Merujuk ke activity_main.xml

        // Inisialisasi Views
        tvTitle = findViewById(R.id.tvTitle) // Pastikan ID ini ada di activity_main.xml
        btnStart = findViewById(R.id.btnStart) // Pastikan ID ini ada di activity_main.xml
        btnToggle = findViewById(R.id.btnToggle) // Pastikan ID ini ada di activity_main.xml
        btnOpenPractice = findViewById(R.id.btnOpenPractice) // Pastikan ID ini ada di activity_main.xml

        // Atur visibilitas awal tombol
        btnToggle.visibility = View.GONE
        btnOpenPractice.visibility = View.GONE

        // Event listener untuk tombol Start
        btnStart.setOnClickListener {
            tvTitle.text = "Aplikasi Dimulai 🚀"
            isHello = false // Perbaikan: Set state agar toggle logic benar setelah start
            btnStart.visibility = View.GONE
            btnToggle.visibility = View.VISIBLE
            btnOpenPractice.visibility = View.VISIBLE
        }

        // Event listener untuk tombol Toggle Pesan
        btnToggle.setOnClickListener {
            if (isHello) {
                tvTitle.text = "Selamat Datang di Mobile Programming 👋"
                btnToggle.text = "Kembalikan Pesan"
            } else {
                tvTitle.text = "Aplikasi Dimulai 🚀" // Perbaikan: Hilangkan spasi ekstra
                btnToggle.text = "Toggle Pesan"
            }
            isHello = !isHello
        }

        // Event listener untuk tombol Buka Latihan Layout
        btnOpenPractice.setOnClickListener {
            // Pastikan LayoutPracticeActivity ada dan dideklarasikan di AndroidManifest.xml
            val intent = Intent(this, LayoutPracticeActivity::class.java)
            startActivity(intent)
        }
    }
}
